### PR TITLE
FlowException can be thrown from within UntrustworthyData.unwrap for …

### PR DESCRIPTION
**Pull request by shamsasari**
…better Java interop, and more checked exception annotations relating to flows

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt [hunk 1]
```diff
@@ -18,11 +18,18 @@ class UntrustworthyData<out T>(private val fromUntrustedWorld: T) {
         @Deprecated("Accessing the untrustworthy data directly without validating it first is a bad idea")
         get() = fromUntrustedWorld
 
-    @Suppress("DEPRECATION")
     @Throws(FlowException::class)
-    inline fun <R> unwrap(validator: (T) -> R) = validator(data)
+    fun <R> unwrap(validator: Validator<T, R>) = validator.validate(fromUntrustedWorld)
```
Comment (mikehearn) [inline core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt]: Why remove inline? The bytecode will still be emitted into the class for Java interop.

File: core/src/main/kotlin/net/corda/flows/FetchDataFlow.kt [hunk 2]
```diff
@@ -7,6 +7,7 @@ import net.corda.core.crypto.SecureHash
 import net.corda.core.flows.FlowException
 import net.corda.core.flows.FlowLogic
 import net.corda.core.utilities.UntrustworthyData
+import net.corda.core.utilities.unwrap
```
Comment (mikehearn) [inline core/src/main/kotlin/net/corda/flows/FetchDataFlow.kt]: Why is this import now needed?

Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/FetchDataFlow.kt]: Kotlin doesn't want to convert a SAM which throws an exception into a lambda, so I had to introduce this extension method.

File: core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt (hunk 1 above)
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt]: inline is only for Kotlin code, right? And this now no longer method isn't Kotlin friendly

Comment (mikehearn) [inline core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt]: Oh, I see. It won't inline through SAM classes, is what you're saying. Perhaps we can just have both versions, or does that hit name conflicts?

Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt]: Java complains about ambiguous method call, and solving it needs an absolutely horrendous cast: 
`.unwrap((UntrustworthyData.Validator<? super SignedTransaction, ? extends SignedTransaction>) stx -> {`
lol

Comment (mikehearn) [inline core/src/main/kotlin/net/corda/core/utilities/UntrustworthyData.kt]: wow. OK then. I admit defeat!

**Pull request merged**